Fixes for troubleshooting guide re. Hubble/Hubble Relay

tklauser opened the pull request:
* Don't mark Hubble Relay as beta in the troubleshooting guide
* Adjust docs now that Hubble is enabled by default and secured via mTLS.
* Fix spelling.
* Fix `..code::` sections to match common style, namely:
  * Use `..code:: shell-session` for sections including commands and their output.
  * Don't use `$` shell prompt in `..code:: bash` sections without output.
  * Fix indentation in `..code::` and `..note::` sections to align.

For #13627

tklauser committed:
docs: no longer mark Hubble Relay as beta in troubleshooting guide

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
docs: fix code and note sections in troubleshooting guide

Follow the common style in our docs:

* Change the .. code:: section from bash to shell-session since it
  contains the command and its output.
* Align content following .. code:: and ..note:: to align properly.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
docs: fix spelling mistakes in Hubble troubleshooting guide

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

rolinh commented on Documentation/operations/troubleshooting.rst:
```diff
@@ -160,76 +160,74 @@ The Hubble CLI is part of the Cilium container image and can be accessed via
+   Max Flows: 4096
+   Current Flows: 2542 (62.06%)
 
 ``cilium-agent`` must be running with the ``--enable-hubble`` option in order
```
```suggestion
``cilium-agent`` must be running with the ``--enable-hubble`` option (default) in order
```

rolinh commented on Documentation/operations/troubleshooting.rst:
```diff
@@ -238,25 +236,21 @@ to set the ``hubble.enabled=true`` value.
-
 Hubble Relay is a service which allows to query multiple Hubble instances
 simultaneously and aggregate the results. As Hubble Relay relies on individual
 Hubble instances, Hubble needs to be enabled when deploying Cilium. In
```
As Hubble is enabled by default in 1.9, this can be simplified a bit:

> Hubble Relay is a service which allows to query multiple Hubble instances simultaneously and aggregate the results. The Hubble service needs to be exposed on TCP port 4244 to allow Hubble Relay to connect to individual instances. This can be done via Helm values  or option ``--hubble-listen-address :4244`` on cilium-agent.

Also, the note below about security implications can be removed as the endpoint is now secured via mTLS by default.

rolinh commented on Documentation/operations/troubleshooting.rst:
```diff
@@ -290,24 +284,24 @@ command:
+   Max Flows: 16384
+   Current Flows: 16384 (100.00%)
 
 For convenience, you may set and export the ``HUBBLE_DEFAULT_SOCKET_PATH``
```
```suggestion
For convenience, you may set and export the ``HUBBLE_SERVER``
```

The old environment variable (`HUBBLE_DEFAULT_SOCKET_PATH`) still works but is deprecated/not recommended anymore.

rolinh commented on Documentation/operations/troubleshooting.rst:
```diff
@@ -290,24 +284,24 @@ command:
 .. code:: bash
 
-    $ export HUBBLE_DEFAULT_SOCKET_PATH=localhost:4245
+   export HUBBLE_DEFAULT_SOCKET_PATH=localhost:4245
```
```suggestion
   export HUBBLE_SERVER=localhost:4245
```

rolinh reviewed (changes requested):
Thanks! While you're at it, this sentence should be removed from the troubleshooting guide as with 1.9 Hubble is enabled by default (#13311):

> It needs to be enabled via the Helm value ``hubble.enabled=true`` or the ``--enable-hubble`` option on cilium-agent.

tklauser committed:
docs: adjust troubleshooting guide to new Hubble defaults

Hubble is enabled by default in 1.9, adjust instructions accordingly.
Also, replace the deprecated HUBBLE_DEFAULT_SOCKET_PATH env var by
HUBBLE_SERVER and remove a note about security implication as the
endpoint is secured via mTLS by default.

Suggested-by: Robin Hahling <robin.hahling@gw-computing.net>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
